src/absil/il.fs

```diff
@@ -1933,6 +1933,8 @@ type ILTypeDef(name: string, attributes: TypeAttributes, layout: ILTypeDefLayout
                extends: ILType option, methods: ILMethodDefs, nestedTypes: ILTypeDefs, fields: ILFieldDefs, methodImpls: ILMethodImplDefs,
                events: ILEventDefs, properties: ILPropertyDefs, securityDeclsStored: ILSecurityDeclsStored, customAttrsStored: ILAttributesStored, metadataIndex: int32) =  
 
+    let customAttrs = lazy (customAttrsStored.GetCustomAttrs metadataIndex)
```

Comment:
If you're going to store these, I suppose it should be `let mutable customAttrsStored = customAttrsStored` and then on access of `CustomAttrs` you replace any `ILAttributesStored.Reader` by `ILAttributesStored.Given` after evaluation.  That won't increase memory size of `ILTypeDef` at all

Thinking it over I think it is now ok to store them for `ILTypeDef`, since memory pressure has been reduced in VS.

Reply:
@dsyme Thank you, done.

Reply:
@dsyme 

> At some point we moved to deliberately not persist the custom attributes because there were so many of them hanging around.

Is there a reason we need to persist all of them versus only storing those that matters to the compiler (only a known set of attribute types are relevant)?

> Now memory pressure is relieved a bit in VS we might be able to persist them
> Thinking it over I think it is now ok to store them for ILTypeDef, since memory pressure has been reduced in VS.

So long FCS is in process, that memory pressure remains a problem, I'd not assume that some of the other optimizations make it ok to just go back on storing all attributes at all times (after first call, but the call is probably done in many cases).

Can we devise a plan to have them tucked away when memory pressure situation occurs?

Can we look at a solution to only store attributes that are relevant to the compilation? this could be done in `ILBinaryReader.customAttrsReader` AFAIU: https://github.com/Microsoft/visualfsharp/blob/cc8bc2b4077711a2efc454edaae2b702f9984e1f/src/absil/ilread.fs#L2554-L2563